src/components/CrossFadeIcon.tsx

```diff
@@ -23,114 +23,100 @@ type Props = {
   theme: ReactNativePaper.Theme;
 };
 
-type State = {
-  currentIcon: IconSource;
-  previousIcon: IconSource | null;
-  fade: Animated.Value;
-};
-
-class CrossFadeIcon extends React.Component<Props, State> {
-  static getDerivedStateFromProps(nextProps: Props, nextState: State) {
-    if (nextState.currentIcon === nextProps.source) {
-      return null;
-    }
-
-    return {
-      currentIcon: nextProps.source,
-      previousIcon: nextState.currentIcon,
-    };
+const CrossFadeIcon = ({ color, size, source, theme }: Props) => {
+  const [currentIcon, setCurrentIcon] = React.useState<IconSource>(
+    () => source
+  );
+  const [previousIcon, setPreviousIcon] = React.useState<IconSource | null>(
+    null
+  );
+  const { current: fade } = React.useRef<Animated.Value>(new Animated.Value(1));
+
+  const { scale } = theme.animation;
+
+  const previousPreviousIconRef = React.useRef<IconSource | null>();
```

Comment:
Can we do:

```suggestion
  const { current: previousPreviousIcon } = React.useRef<IconSource | null>();
```

and remove line 41?

Reply:
After changing it to 
```
  let { current: previousPreviousIcon } = React.useRef<IconSource | null>();
  React.useEffect(() => {
    previousPreviousIcon = previousIcon;
  }, [previousIcon]);
```
and removing line 41

Im getting a warning from eslint:
ESLint: Assignments to the 'previousPreviousIcon' variable from inside React Hook React.useEffect will be lost after each render. To preserve the value over time, store it in a useRef Hook and keep the mutable value in the '.current' property. Otherwise, you can move this variable directly inside React.useEffect.(react-hooks/exhaustive-deps)

Reply:
Ok then let's keep the ref as it is now, but let's remove line 41 anyway, since `previousPreviousIcon ` is used only once so there's no need to make it and we can directly use `previousPreviousIconRef.current`

Reply:
Sure